pages/blog/zenwave-asyncapi-code-generator.md

```diff
@@ -0,0 +1,348 @@
+```
+
+
+### Spring Cloud Streams Consumer: Using generated code to consume messages
+
+On the consumer side generates:
+
+- Functional Consumer `DoCustomerRequestConsumer`  for Spring Cloud Streams bindings.
+- Business Interface `IDoCustomerRequestConsumerService` you need to implement in order to receive strongly typed messages.
+
+This Functional Consumer can abstract away different integration patterns like Business Dead Letter Queue and others... depending on how you configure zenwave maven generator.
+
+**To consume messages you need to implement generated business interface and register it as a Spring bean.**
+
+```java
+// Autogenerated: you need to implement and provide this business interface
+public interface IOnCustomerEventConsumerService {
+    /**
+     * Customer Domain Events
+     */
+    default void onCustomerEvent(CustomerEventPayload payload, CustomerEventPayloadHeaders headers) {};
+}
+```
+
+```java
+// Autogenerated: add it to your autoscan packages and provide business interface implementation
+@Component("on-customer-event")
+public class OnCustomerEventConsumer implements Consumer<Message<CustomerEventPayload>> {
+
+    // you need to implement this interface
+    protected IOnCustomerEventConsumerService service;
+
+    @Override
+    public void accept(Message<CustomerEventPayload> message) {
+        log.debug("Received message: {}", message);
+        try {
+            Object payload = message.getPayload();
+            if (payload instanceof CustomerEventPayload) {
+                var headers = new IOnCustomerEventConsumerService.CustomerEventPayloadHeaders();
+                headers.putAll(message.getHeaders());
+                service.onCustomerEvent((CustomerEventPayload) payload, headers);
+                return;
+            }
+            log.error("Received message without any business handler: [payload: {}, message: {}]", payload.getClass().getName(), message);
+        } catch (Exception e) {
+            // error handling and dead-letter-queue routing omitted for brevity
+        }
+    }
+}
+```
+
+```java
+// Implement the business interface and add it to your context
+@Component
+class DoCustomerRequestConsumerService implements IDoCustomerRequestConsumerService {
+
+    @Override
+    public void doCustomerRequest(CustomerRequestPayload payload, CustomerRequestPayloadHeaders headers) {
+        log.info("Received '{}' message with payload: {}", payload.getClass(), payload);
+        // [...] do something with this message
+    }
+}
+```
+
+### Exception Handling with Business Dead Letter Queue
+
+ZenWave SDK consumers can be configured to route exceptions to different error queues. This is useful to manage non-retryable business exceptions so the stream processing is not interrupted. If your code throws an exception not configured for error routing it will be rethrow and it will follow the standard error handling mechanism for your particular Spring Cloud Stream binder.
```

Comment:
```suggestion
ZenWave SDK consumers can be configured to route exceptions to different error queues. This is useful to manage non-retrayable business exceptions so the stream processing is not interrupted. If your code throws an exception not configured for error routing, it will be rethrown and follow the standard error handling mechanism for your particular Spring Cloud Stream binder.
```